backend/api/schedule/types.py

```diff
@@ -51,3 +54,78 @@ def image(self, info) -> Optional[str]:
             return None
 
         return info.context.request.build_absolute_uri(self.image.url)
+
+
+@strawberry.type
+class ScheduleInvitationDate:
+    id: strawberry.ID
+    start: datetime
+    end: datetime
+
+    @classmethod
+    def from_django(cls, schedule_item):
+        hour_slot = schedule_item.slot.hour
+        day = schedule_item.slot.day.day
+        start = datetime.combine(day, hour_slot)
+        duration = schedule_item.duration or schedule_item.slot.duration
+        return cls(
+            id=schedule_item.id, start=start, end=start + timedelta(minutes=duration)
+        )
+
+
+@strawberry.enum
+class ScheduleInvitationOption(Enum):
+    NO_ANSWER = "no_answer"
+    CONFIRM = "confirm"
+    MAYBE = "maybe"
+    REJECT = "reject"
+    CANT_ATTEND = "cant_attend"
+
+    def to_schedule_item_status(self):
+        if self == ScheduleInvitationOption.CONFIRM:
+            return ScheduleItemModel.STATUS.confirmed
+
+        if self == ScheduleInvitationOption.MAYBE:
+            return ScheduleItemModel.STATUS.maybe
+
+        if self == ScheduleInvitationOption.REJECT:
+            return ScheduleItemModel.STATUS.rejected
+
+        if self == ScheduleInvitationOption.CANT_ATTEND:
```

Comment:
It's missing `NO_ANSWER` I think... 
maybe a dictionary mapping the statuses would be nicer :)

Reply:
I didn't want to add NO_ANSWER since it isn't a valid response, but I will add it in case it causes confusion 👍  a dict is ok too, I wanted to be explicit but doesn't matter